### PR TITLE
DLC Wallet pulldown

### PR DESCRIPTION
**Pull request by benthecarman**

### CODE REVIEW

File: app/server-test/src/test/scala/org/bitcoins/server/RoutesSpec.scala [hunk 1]
```diff
@@ -13,44 +12,55 @@ import org.bitcoins.core.config.RegTest
...
 import scala.collection.mutable
+import scala.concurrent.duration.DurationInt
 import scala.concurrent.{ExecutionContext, Future}
 
 class RoutesSpec extends AnyWordSpec with ScalatestRouteTest with MockFactory {
 
   implicit val conf: BitcoinSAppConfig =
     BitcoinSTestAppConfig.getSpvTestConfig()
 
+  implicit val timeout: RouteTestTimeout = RouteTestTimeout(5.seconds)
```
Comment (nkohen) [inline app/server-test/src/test/scala/org/bitcoins/server/RoutesSpec.scala:62]: I forget what the rationale is for this? Was 10 too much for some reason?

File: dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala [hunk 2]
```diff
@@ -0,0 +1,1374 @@
...
+import org.bitcoins.wallet.{Wallet, WalletLogger}
+import scodec.bits.ByteVector
+
+import java.time.Instant
+import scala.concurrent.{ExecutionContext, Future}
+
+/** A [[Wallet]] with full DLC Functionality */
+abstract class DLCWallet
+    extends Wallet
+    with AnyDLCHDWalletApi
```
Comment (nkohen) [inline dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala:39]: Could you file an issue detailing the proposed follow-up wallet refactor?

File: db-commons/src/main/resources/reference.conf [hunk 3]
```diff
@@ -117,6 +117,22 @@ bitcoin-s {
...
+  dlc = ${bitcoin-s.dbDefault}
+  dlc {
+    # this config key is read by Slick
+    db {
+      name = dlcdb
+      url = "jdbc:sqlite:"${bitcoin-s.dlc.db.path}${bitcoin-s.dlc.db.name}
+    }
+    # PostgreSQL example:
+    # db {
+    #   url = "jdbc:postgresql://localhost:5432/dlc"
```
Comment (nkohen) [inline db-commons/src/main/resources/reference.conf]: shouldn't this be `dlcdb`?

File: db-commons/src/main/scala/org/bitcoins/db/CRUD.scala [hunk 4]
```diff
@@ -84,21 +84,16 @@ abstract class CRUD[T, PrimaryKeyType](implicit
...
       for {
         _ <- safeDatabase.runVec(DBIO.sequence(actions).transactionally)
         result <- safeDatabase.runVec(findAll(ts).result)
       } yield result
     }
-
-    FutureUtil.foldLeftAsync(Vector.empty[T], ts) { (accum, t) =>
-      oldUpdateAll(Vector(t)).map(accum ++ _)
-    }
   }
```
Comment (nkohen) [inline db-commons/src/main/scala/org/bitcoins/db/CRUD.scala:97]: Ping @Christewart would be nice to have extra 👀  on this

File: dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala [hunk 5]
```diff
@@ -0,0 +1,1374 @@
...
+  private[bitcoins] val dlcSigsDAO: DLCCETSignaturesDAO = DLCCETSignaturesDAO()
+  private[bitcoins] val dlcRefundSigDAO: DLCRefundSigsDAO = DLCRefundSigsDAO()
+  private[bitcoins] val remoteTxDAO: DLCRemoteTxDAO = DLCRemoteTxDAO()
+
+  private def calcContractId(offer: DLCOffer, accept: DLCAccept): ByteVector = {
+    val builder = DLCTxBuilder(offer, accept.withoutSigs)
+    builder.calcContractId
+  }
+
+  private def updateDLCContractIds(
```
Comment (nkohen) [inline dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala:68]: scaladocs for people not familiar with this codebase on these defs (including the ones below) would be nice

File: app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala [hunk 6]
```diff
@@ -161,6 +162,32 @@ object BitcoindRpcBackendUtil extends Logging {
     }
   }
 
+  def createDLCWalletWithBitcoindCallbacks(
+      bitcoind: BitcoindRpcClient,
+      wallet: DLCWallet)(implicit system: ActorSystem): DLCWallet = {
+    // Kill the old wallet
+    wallet.stop()
```
Comment (Christewart) [inline app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala]: I don't think this is necessary now. This is a noop . All wallet resources were moved into `WalletAppConfig`

https://github.com/bitcoin-s/bitcoin-s/blob/4c1b5e345242afa6c1b5a9298f0c9cced94fa73f/wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala#L156

File: db-commons/src/main/scala/org/bitcoins/db/CRUD.scala (hunk 4 above)
Comment (Christewart) [reply db-commons/src/main/scala/org/bitcoins/db/CRUD.scala:97]: It looks right to me, hopefully there is unit tests to confirm that

**Review by Christewart — APPROVED** [review]
We just needs to get this in at this point, we can iterate and improve once in master

File: db-commons/src/main/scala/org/bitcoins/db/CRUD.scala (hunk 4 above)
Comment (benthecarman) [reply db-commons/src/main/scala/org/bitcoins/db/CRUD.scala:97]: We do have tests :)

File: app/server-test/src/test/scala/org/bitcoins/server/RoutesSpec.scala (hunk 1 above)
Comment (benthecarman) [reply app/server-test/src/test/scala/org/bitcoins/server/RoutesSpec.scala:62]: I forget why as well. It works without it tho

File: db-commons/src/main/resources/reference.conf (hunk 3 above)
Comment (benthecarman) [reply db-commons/src/main/resources/reference.conf]: We actually don't need the url anymore, fixed this

File: dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala (hunk 2 above)
Comment (benthecarman) [reply dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala:39]: #3139

**Pull request merged**